packages/backend/src/ee/database/migrations/20250516162139_add_ai_agents_table.ts

```diff
@@ -0,0 +1,51 @@
+import { Knex } from 'knex';
+
+const AIAgentTableName = 'ai_agent';
+
+export async function up(knex: Knex): Promise<void> {
+    if (await knex.schema.hasTable(AIAgentTableName)) return;
+
+    await knex.schema.createTable(AIAgentTableName, (table) => {
+        table
+            .uuid('ai_agent_uuid')
+            .primary()
+            .defaultTo(knex.raw('uuid_generate_v4()'));
+
+        table
+            .uuid('project_uuid')
+            .notNullable()
+            .references('project_uuid')
+            .inTable('projects')
+            .onDelete('CASCADE');
+        table
+            .uuid('organization_uuid')
+            .notNullable()
+            .references('organization_uuid')
+            .inTable('organizations')
+            .onDelete('CASCADE');
+
+        table.string('name').notNullable();
+        table.text('description');
+        table.text('image_url');
+        table.jsonb('data_sources').defaultTo({ fieldNameTags: null });
```

Comment:
This should ideally be avoided, as it’s equivalent to using `any` in TS and makes data migrations difficult. Based on our recent conversation with the customers, this can be either NULL (meaning all explores and fields are available) or an nonempty array of tags that will filter both explores and fields based on the tags.